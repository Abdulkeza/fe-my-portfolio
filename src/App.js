import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
	decrement,
	increment,
	incrementByAmount,
} from './redux/reducers/counter.js';

function App() {
	// we are distructuring "count" from counter.js as 'initial state' (from "store.js" to ("rootReducer.js" -counter )to counter.js )
	const { count } = useSelector((state) => state.allReducers.counter);
	const dispatch = useDispatch();

	return (
		<>
			<h1>Hi here! This is Keza-codes back</h1>
			<div>
				<div>
					<button type="button" onClick={() => dispatch(increment())}>
						Increment
					</button>
					<h2>
						<span>value: {count}</span>
					</h2>

					<button type="button" onClick={() => dispatch(decrement())}>
						Decrement
					</button>

					<button type="button" onClick={() => dispatch(incrementByAmount(10))}>
						Increment by 10
					</button>
				</div>
			</div>
		</>
	);
}

export default App;
