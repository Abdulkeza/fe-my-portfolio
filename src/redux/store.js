import { configureStore } from '@reduxjs/toolkit';
import allReducers from './rootReducer.js';

export default configureStore({
	reducer: {
		allReducers,
	},
});
