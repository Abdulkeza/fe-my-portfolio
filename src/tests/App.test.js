import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import App from '../App.js';
import store from '../redux/store.js';

describe('Tests for App', () => {
	test('Should render App component', () => {
		const wrapper = render(
			<Provider store={store}>
				<App />
			</Provider>
		);
		expect(wrapper).toMatchSnapshot();
	});
});

describe('Testing App elements', () => {
	beforeEach(() => {
		render(
			<Provider store={store}>
				<App />
			</Provider>
		);
	});

	test('Should find h1', () => {
		const h1 = screen.getByText(/Hi here! This is Keza-codes back/i);
		expect(h1).toBeInTheDocument();
	});

	test('Should find h1 elements', () => {
		const h1 = screen.getByRole('heading', { level: 1 });
		expect(h1).toBeInTheDocument();
	});

	test('Should find button elements', () => {
		const button = screen.getByRole('button', { name: 'Increment' });
		expect(button).toBeInTheDocument();
	});

	describe('Testing functionalities', () => {
		test('should return initial value as 0', () => {
			const value = screen.getByText(/value: 0/i);
			expect(value).toBeInTheDocument();
		});

		test('should increment on increment button', () => {
			const incrementBtn = screen.getAllByText(/Increment/i)[0];
			fireEvent.click(incrementBtn);
			const value = screen.getByText(/value: 1/i);
			expect(value).toBeInTheDocument();
		});

		test('should decrement on Decrement button after incremented by 1', () => {
			const decrementBtn = screen.getByText(/Decrement/i);
			fireEvent.click(decrementBtn);
			const value = screen.getByText(/value: 0/i);
			expect(value).toBeInTheDocument();
		});

		test('should increment by 10 on Increment by 10 button', () => {
			const incrementBtnBy10 = screen.getByText(/Increment by 10/i);
			fireEvent.click(incrementBtnBy10);
			const value = screen.getByText(/value: 10/i);
			expect(value).toBeInTheDocument();
		});
	});
});
