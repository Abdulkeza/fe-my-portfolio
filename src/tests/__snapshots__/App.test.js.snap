// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for App Should render App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Hi here! This is Keza-codes back
      </h1>
      <div>
        <div>
          <button
            type="button"
          >
            Increment
          </button>
          <h2>
            <span>
              value: 
              0
            </span>
          </h2>
          <button
            type="button"
          >
            Decrement
          </button>
          <button
            type="button"
          >
            Increment by 10
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h1>
      Hi here! This is Keza-codes back
    </h1>
    <div>
      <div>
        <button
          type="button"
        >
          Increment
        </button>
        <h2>
          <span>
            value: 
            0
          </span>
        </h2>
        <button
          type="button"
        >
          Decrement
        </button>
        <button
          type="button"
        >
          Increment by 10
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
